manifests/init.pp

```diff
@@ -7,9 +7,9 @@
   Hash $common_config = {},
   String $libdir,
   String $configdir,
-  String $plugin_owner,
-  String $plugin_group,
-  String $plugin_mode,
+  Optional[String] $plugin_owner,
+  Optional[String] $plugin_group,
+  Optional[String] $plugin_mode,
```

Comment:
I'm not sure what conditionals you're referring to? I'm pretty sure `String` can't be `undef`, which is why I made these `Optional`.